### ISSUE TITLE
refactor(solvers): `solve_cg` as a thin shim over `gaussx.linear_solve(CGSolver, OperatorPreconditioner)`; expose `MultigridSolver` as a gaussx `AbstractPreconditioner` / lineax solver

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#316]
Part of #279. Fixes the diagnostics/gradient defects of #307 as a side effect.

## Summary

`solve_cg` (`finitevolx/_src/solvers/iterative.py:48-143`) wraps `lineax.CG` with a sign flip and re-wraps the preconditioner as a callable through `options`. gaussx (already at the pinned v0.0.17) provides the same thing as first-class objects, differentiably:

```python
op = gaussx.as_linear_operator(A_flat, shape=(n, n), negative_definite=True)
x = gaussx.linear_solve(op, b, solver=gaussx.CGSolver(rtol=1e-6, atol=1e-6, max_steps=500),
                        preconditioner=gaussx.OperatorPreconditioner(lambda v: -pc(v)))
```

Benchmark, 128² circular mask, λ = 4, FFT preconditioner, rtol = atol = 1e-6:

```
finitevolx.solve_cg:                16.61 ms   rel.res = 4.48e-08
gaussx.linear_solve(CGSolver):      17.55 ms   rel.res = 4.48e-08   |x1 − x2| = 0.00e+00
gaussx.linear_solve (2-D structure): 18.61 ms
```

`solve_cg` is fully redundant; the only thing it adds is `CGInfo`, which is unreliable (#307). `MultigridSolver` (~1000 lines, genuinely finitevolX-specific) is only usable through its own `v_cycle` and `make_multigrid_preconditioner` closure; as a gaussx `AbstractPreconditioner` it would compose with `gaussx.linear_solve`, `MINRESSolver`, etc.

## Proposed fix

1. `solve_cg` becomes: build the flat operator from the 2-D matvec, call `gaussx.linear_solve`, reshape; `CGInfo` from the lineax result (`result == successful`, explicit final residual). Keep the signature.
2. `MultigridSolver.as_preconditioner() -> gaussx.AbstractPreconditioner` (one V-cycle) and `as_lineax_solver()` (V-cycles to tolerance), after the multigrid fixes in #277.
3. `masked_laplacian` gains a `.as_operator()` returning a lineax `FunctionLinearOperator` with the negative-definite tag.
4. Bump the gaussx pin to v0.0.28 (signatures identical; 0 removals).

## Acceptance criteria

- [ ] `solve_cg` results bit-identical to today on the existing tests; `converged=False` on `max_steps`.
- [ ] `jax.grad` through `solve_cg` w.r.t. λ and dx with a preconditioner matches finite differences.
- [ ] `gaussx.linear_solve(op, b, solver=CGSolver(), preconditioner=mg.as_preconditioner())` converges in ≤ 15 iterations on the 128² basin (after #305).
